### PR TITLE
Do not override SPOTIFY_JSON_BUILD_TESTS

### PR DESCRIPTION
**Pull request by 8W9aG**
* Allow people adding spotify-json as a subdirectory
to specify whether to build the tests

### CODE REVIEW

**Comment by 8W9aG** [comment]
@punchfox @fxb @pereckerdal

File: CMakeLists.txt [hunk 1]
```diff
@@ -136,7 +136,9 @@ set(double_conversion_INCLUDE_DIR ${CMAKE_CURRENT_LIST_DIR}/vendor/double-conver
 target_include_directories(${json_library_TARGET} PUBLIC ${double_conversion_INCLUDE_DIR})
 target_link_libraries(${json_library_TARGET} double-conversion)
 
-option(SPOTIFY_JSON_BUILD_TESTS "Build tests and benchmarks" ON)
+if (NOT DEFINED SPOTIFY_JSON_BUILD_TESTS)
```
Comment (punchfox) [inline CMakeLists.txt]: Is it not possible to set options from the including CMake file?

Also, our CMake style says no space between `if` and `(`.

File: CMakeLists.txt [hunk 2]
```diff
@@ -136,7 +136,9 @@ set(double_conversion_INCLUDE_DIR ${CMAKE_CURRENT_LIST_DIR}/vendor/double-conver
...
+  option(SPOTIFY_JSON_BUILD_TESTS "Build tests and benchmarks" ON)
+endif ()
```
Comment (punchfox) [inline CMakeLists.txt]: `endif()`

**Comment by fxb** [comment]
You need to override it by specifying ```CACHE```, then it will work.

Try this for example:

```set(SPOTIFY_JSON_BUILD_TESTS OFF CACHE BOOL "Don't build spotify-json tests" FORCE)```

**Comment by 8W9aG** [comment]
Ah thanks I will do that

**Pull request closed without merging**